### ISSUE TITLE
您好，请问DDP模式下，如何将不同GPU上backbone提取的特征向量汇聚到一起来计算损失

### ISSUE THREAD

**Issue by feifeifeiliu** [JDAI-CV/fast-reid#490]
例如，三元损失需要一定大小的batchsize以构成足够数量的三元组。而据我所知，DDP会在每张GPU上单独计算这个GPU上的样本的损失，这样就大大减少了三元组的数量。我尝试了使用
`
def distributed_concat(tensor):
    output_tensors = [tensor.clone() for _ in range(torch.distributed.get_world_size())]
    torch.distributed.all_gather(output_tensors, tensor)
    concat = torch.cat(output_tensors, dim=0)
    return concat

pred_features = distributed_concat(pred_features)
gt_labels = distributed_concat(gt_labels)
`
如上的代码，期望将各个GPU上的pred_features汇聚到一起计算损失，结果是损失根本不能正常下降。非常期待您的帮助和解答！

**Comment by L1aoXingyu** [comment]
你可以直接参考我们注掉的部分
https://github.com/JDAI-CV/fast-reid/blob/ff8a958fff85f56f6892d28e1f65e662650cef6d/fastreid/modeling/losses/triplet_loss.py#L86

**Comment by feifeifeiliu** [comment]
谢谢您的解答~
